scripts/reboot

```diff
@@ -242,6 +253,23 @@ if [ -x ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE} ]; then
     ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE}
 fi
 
+if [ -x ${DEVPATH}/${PLATFORM}/${PRE_REBOOT_HOOK} ]; then
```

Comment:
@vadymhlushko-mlnx why do we need pre_reboot_hook?

Reply:
@prgeor this PR comes in bundle [[graceful reboot] Rename the platform_reboot to the pre_reboot_hook, remove the sysfs power cycle #18324](https://github.com/sonic-net/sonic-buildimage/pull/18324).

For MLNX rename the `platform_reboot` script to `pre_reboot_hook` because we remove the `sysfs power cycle` from `platform